plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnHandle.java

```diff
@@ -148,44 +180,57 @@ public long getRetainedSizeInBytes()
     @Override
     public int hashCode()
     {
-        return Objects.hash(baseColumnName, baseType, fieldId, physicalName, physicalType, columnType);
+        return Objects.hash(baseColumnName, baseType, fieldId, physicalName, physicalType, columnType, baseColumnIndex, projectionInfo);
     }
 
     @Override
     public String toString()
     {
-        return baseColumnName + ":" + baseType.getDisplayName() + ":" + columnType;
+        return getQualifiedName() +
+                ":" + (projectionInfo.isEmpty() ? baseType.getDisplayName() : projectionInfo.get().getType().getDisplayName()) +
+                ":" + columnType;
     }
 
     public HiveColumnHandle toHiveColumnHandle()
     {
+        return toHiveColumnHandle(physicalName); // this name is used for accessing Parquet files, so it should be physical name
+    }
+
+    public HiveColumnHandle toHiveColumnHandle(String baseColumnName)
+    {
+        Optional<HiveColumnProjectionInfo> hiveColumnProjectionInfo = projectionInfo.map(deltaLakeColumnProjectionInfo -> new HiveColumnProjectionInfo(
+                deltaLakeColumnProjectionInfo.getDereferenceIndices(),
+                deltaLakeColumnProjectionInfo.getDereferenceNames(),
+                toHiveType(deltaLakeColumnProjectionInfo.getType()),
+                deltaLakeColumnProjectionInfo.getType()));
+
         return new HiveColumnHandle(
-                physicalName, // this name is used for accessing Parquet files, so it should be physical name
-                0, // hiveColumnIndex; we provide fake value because we always find columns by name
```

Comment:
Wasn't that an important comment?

We still always find columns by name.
Do we have tests where file schema has different column order (or certain fields missing) from the table schema?

We shouldn't need the `baseColumnIndex` in the column handle.

Reply:
While creating the [Parquet page source](https://github.com/trinodb/trino/blob/master/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java#L203) for the Delta table, It basically projects [base columns](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSourceProvider.java#L712).

The current implementation of `projectBaseColumns` projects based on `baseHiveColumnIndex` if there is at least one non-base-column. That is why I needed to provide `hiveColumnIndex` when converting `DeltaLakeColumnHandle` to `HveColumnHandle`.

I am looking to modify `HivePageSourceProvider#projectBaseColumns` to project  base column based on `baseColumnName` as well.

Reply:
Added test case `TestDeltaLakeConnectorTest #testProjectionPushdownColumnReorderInSchemaAndDataFile`